usaspending_api/broker/management/commands/fabs_nightly_loader.py

```diff
@@ -0,0 +1,305 @@
+            awarding_agency = Agency.get_by_toptier_subtier(
+                row['awarding_agency_code'],
+                row["awarding_sub_tier_agency_c"]
+            )
+            created, award = Award.get_or_create_summary_award(
+                awarding_agency=awarding_agency,
+                piid=row.get('piid'),
+                fain=row.get('fain'),
+                uri=row.get('uri'),
+                parent_award_id=row.get('parent_award_id'))
+            award.save()
+
+            award_update_id_list.append(award.id)
+
+            parent_txn_value_map = {
+                "award": award,
+                "awarding_agency": awarding_agency,
+                "funding_agency": Agency.get_by_toptier_subtier(row['funding_agency_code'],
+                                                                row["funding_sub_tier_agency_co"]),
+                "recipient": legal_entity,
+                "place_of_performance": pop_location,
+                "period_of_performance_start_date": format_date(row['period_of_performance_star']),
+                "period_of_performance_current_end_date": format_date(row['period_of_performance_curr']),
+                "action_date": format_date(row['action_date']),
+            }
+
+            fad_field_map = {
+                "type": "assistance_type",
+                "description": "award_description",
+            }
+
+            transaction_dict = load_data_into_model(
+                TransactionNormalized(),  # thrown away
+                row,
+                field_map=fad_field_map,
+                value_map=parent_txn_value_map,
+                as_dict=True)
+
+            transaction = TransactionNormalized.get_or_create_transaction(**transaction_dict)
+            transaction.save()
+
+            financial_assistance_data = load_data_into_model(
+                TransactionFABS(),  # thrown away
+                row,
+                as_dict=True)
+
+            afa_generated_unique = financial_assistance_data['afa_generated_unique']
+
+            if TransactionFABS.objects.filter(afa_generated_unique=afa_generated_unique).exists():
+                TransactionFABS.objects.filter(afa_generated_unique=afa_generated_unique).\
+                    update(**financial_assistance_data)
+            else:
+                transaction_assistance = TransactionFABS(transaction=transaction, **financial_assistance_data)
+                transaction_assistance.save()
+
+    def add_arguments(self, parser):
+
+        parser.add_argument(
+            '--date',
+            dest="date",
+            nargs='+',
+            type=str,
+            help="(OPTIONAL) Date from which to start the nightly loader. Expected format: MM/DD/YYYY"
+        )
+
+    @transaction.atomic
+    def handle(self, *args, **options):
+        logger.info('Starting historical data load...')
+
+        db_cursor = connections['data_broker'].cursor()
```

Comment:
Curious to know the reasoning for creating the connection cursor here, only to be used in `get_fabs_data`

Reply:
that can be moved into `get_fabs_data`. This is likely a result from being copied from previous efforts where the cursor was used more than it is currently.